source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+The following sections use a ``cursor`` variable, which is a cursor that
+retrieved all the documents in a collection with the ``Find()`` function.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+Retrieve All Documents
+----------------------
+
+To populate an array with all of your query results, use the ``All()``
+method:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor all
+   :end-before: end cursor all
+
+.. important:: Memory
+
+   This method crashes your application if your query retrieves a number
+   of documents larger than the size of your array. If this occurs,
+   :ref:`retrieve the documents individually <individual-documents-golang>`.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually, use the
+``Next()`` method.
+
+This method blocks the current goroutine. That means the function
+returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. tip::
+
+   Iterating through your documents individually removes the possibility
+   of crashing your application and allows you to :ref:`access a cursors
+   elements <cursor-elements-golang>`.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from your cursor, use the ``TryNext()``
+method.
+
+This method is unable to block the current goroutine. That means the
+function returns a document if each of the following is true:
+
+- A document is currently available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor try next
+   :end-before: end cursor try next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+.. _cursor-elements-golang:
+
+Access Cursor Elements
+----------------------
```

Comment:
Suggestion: Can this section be deleted? I believe it largely duplicates the content of the Go API Documentation, and takes up significant real-estate on this page.

As an alternative, could the reader be linked to the [Working with BSON page](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13833-CRUDCursor/fundamentals/bson/#unmarshalling) page earlier in the document when the `Decode()` method is used in the `Next()` example? 

I think `Current` and the "Other Elements" can be left to the API documentation, as why a reader might be interested in them is not discussed here.